backend/src/main/java/ca/bc/gov/hlth/mohums/controller/UsersController.java

```diff
@@ -54,14 +67,122 @@ public UsersController(WebClientService webClientService, @Value("${config.vanit
         username.ifPresent(usernameValue -> queryParams.add("username", usernameValue));
 
         ResponseEntity<List<Object>> searchResults = webClientService.getUsers(queryParams);
+        
         List<Object> users = searchResults.getBody();
-
+        logger.info("size for users - 1st search: " + users.size());
+        
         if (org.isPresent() && !CollectionUtils.isEmpty(users)) {
             List<Object> filteredUsers = users.stream().filter(new FilterUserByOrgId(org.get())).collect(Collectors.toList());
+            users = filteredUsers;
             
+            logger.debug("size for filteredUsers by org: " + filteredUsers.size());
             searchResults = ResponseEntity.status(searchResults.getStatusCode()).body(filteredUsers);
         }
-
+        
+        // if LastLog param are present then run the Events search to find lastLogDate for each user
+       Integer valueIteration=100;
+       if ((lastLogAfter.isPresent() || lastLogBefore.isPresent()) && !CollectionUtils.isEmpty(users)) {
+            
+            // 1st option: for every user, query all their events => NO, could 1000s of users
+            // or 2nd option: get all events for last year and filter by date and users
+        
+            logger.debug("Start of search+filter all events by user for last Log Date: " + LocalDateTime.now());
+            
+            List<Object> allEventsLastLog= new ArrayList <> ();
+            // Type=LOGIN and dateFrom=one year from now
+            LocalDate oneYearAgo = LocalDate.now().minus(1, ChronoUnit.YEARS);
+            Optional<String> oneYearAgoParam = Optional.of(oneYearAgo.format(DateTimeFormatter.ofPattern("yyyy-MM-dd")));
+            
+            MultiValueMap<String, String> queryEventLastLogParams = buildQueryEventActiveParam("0", valueIteration.toString(), oneYearAgoParam, Optional.empty());
+            List<Object> eventsLastLog = (ArrayList) webClientService.getEvents(queryEventLastLogParams).getBody();
+            
+            Integer start = 0;
+            while(eventsLastLog != null && !eventsLastLog.isEmpty()){
+            
+                allEventsLastLog.addAll(eventsLastLog);
+                
+                if(eventsLastLog.size() == valueIteration){
+                    start = start+valueIteration;
+                    queryEventLastLogParams = buildQueryEventActiveParam(start.toString(), valueIteration.toString() , lastLogAfter, lastLogBefore);
```

Comment:
This block of code, from line 91 to 112, is loading all login events form the past year, right?

So on line, 96, we start at event 0, go up to 100, for one year ago.

```java
MultiValueMap<String, String> queryEventLastLogParams = buildQueryEventActiveParam("0", valueIteration.toString(), oneYearAgoParam, Optional.empty());
```

Maybe maybe there was more than 100 log events, so we need to get more:

```java
queryEventLastLogParams = buildQueryEventActiveParam(start.toString(), valueIteration.toString() , lastLogAfter, lastLogBefore);
```

I see we've updated the `start` parameter here, as expected, but why did we update the `dateFrom` and `dateTo` parameters? Shouldn't they stay the same if we're still just looking for the "one year" range?

Reply:
yes, you are right, the same event should run over and over. It's typo/copy/paste error. I'll fix it in next commit.